src/components/habits/habits.js

```diff
@@ -104,20 +74,23 @@ export default class HabitsList extends React.Component {
 
 
     render (){
-        let allTheThings = [];
-        for (var i = 0; i < this.state.habitslist.length; i++) {
-            allTheThings.push(<Item message={this.state.habitslist[i].title} />);
-        }
-        let items = allTheThings.map(thing => thing);
-
-        return (
-
+        let items = this.state.habitslist.map((habit, i) => (<Item key={i} message={habit.title} />))
+        return(
             <div className='habits-list'>
                 <h2>Habits List</h2>
                 <h4>{items}</h4>
-                <AddHabit  onAdd={(e) => this.addNewHabit(e)} ></AddHabit>
+                <AddHabit onAdd={(e) => this.addNewHabit(e)} ></AddHabit>
             </div>
         );
     }
 }
 
+const mapStateProps = () => ({
+});
+
+const mapDispatchToProps = (dispatch) => ({
+    fetchHabitslist: () => dispatch(fetchHabitslist()),
```

Comment:
why fetching habits is part of redux, but adding new one or checking existing is not?

Reply:
also this thing is never called, why?